test/Test.hs

```diff
@@ -1,12 +1,34 @@
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE FlexibleContexts #-}
 
 module Main where
 
-import Test.Tasty
-import Test.Tasty.HUnit
+import           Test.Tasty
+import           Test.Tasty.HUnit
+
+import           Control.Exception
+import           Control.Monad.Except
+import           Control.Monad.Identity
+import           Control.Monad.Logic
+import           Control.Monad.Reader
+import qualified Control.Monad.State.Lazy as SL
+import qualified Control.Monad.State.Strict as SS
+import           Data.Maybe
+
+#if MIN_VERSION_base(4,9,0)
```

Comment:
Use `MIN_VERSION_tasty_expected_failure` to guard `Test.Tasty.ExpectedFailure`, and `MIN_VERSION_base` for other incompatibilities.